### PR TITLE
Corrections

### PR DESCRIPTION
**Pull request by BBCMarkMcDonnell**
## Problem

Noticed some inconsistencies while reviewing overall framework.

### CODE REVIEW

File: lib/alephant/renderer/view_mapper.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require_relative './error/invalid_path'
```
Comment (revett) [inline lib/alephant/renderer/view_mapper.rb]: `require_relative` need to be used here?

File: spec/renderer_spec.rb [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
   }
-  let(:data) { {content: 'test'} } 
+  let(:data) { {content: 'test'} }
 
   describe '.create' do
     context 'using valid params' do
       let(:expected) { Alephant::Renderer::Renderer }
 
-      specify { 
+      specify {
```
Comment (revett) [inline spec/renderer_spec.rb]: `do` ... `end`

File: spec/renderer_spec.rb [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-  let(:config) { 
+  let(:config) {
     {
-      renderer_id: 'foo',
+      renderer_id: 'foo-renderer',
       view_path: File.join(File.dirname(__FILE__), 'fixtures/components'),
     }
   }
-  let(:data) { {content: 'test'} } 
+  let(:data) { {content: 'test'} }
```
Comment (revett) [inline spec/renderer_spec.rb]: spaces around:

```ruby
let(:data) { { content: 'test' } }
```

File: spec/renderer_spec.rb [hunk 4]
```diff
@@ -1,21 +1,21 @@
 require 'spec_helper'
 
 describe Alephant::Renderer do
-  let(:config) { 
+  let(:config) {
     {
-      renderer_id: 'foo',
+      renderer_id: 'foo-renderer',
```
Comment (revett) [inline spec/renderer_spec.rb]: Double quotes?

File: lib/alephant/renderer/view_mapper.rb (hunk 1 above)
Comment (revett) [inline lib/alephant/renderer/view_mapper.rb]: `require 'alephant/renderer/error/invalid_path'`

**Comment by revett** [comment]
:cake: after some small comments

Lots of single quotes if you want to run **Rubo** over it.

**Comment by BBCMarkMcDonnell** [comment]
@revett run rubocop over all the codez and the specs still pass

**Comment by revett** [comment]
:cake:

**Comment by samfrench** [comment]
Hi @Integralist, myself and @leostaples are getting an error in alephant preview now.

```
NoMethodError at /component/council_editorial_text
undefined method `new' for nil:NilClass
file: view_mapper.rb location: model line: 49
```

Can you take a look please?

**Comment by Integralist** [comment]
@samfrench There's been no major changes in this PR that would cause that type of error. If Alephant Preview is currently utilising the version `1.0.4` then that's quite an old version considering that was released Apr 24 and since then we've had a minor release followed by a major release to 2.0 and two patch releases following that, so I assume there are some big code differences between versions 1 and 2

**Comment by samfrench** [comment]
Looking at the versions, the latest version of 2.0.2 is broken for us, but 2.0.1 works.

I created this branch, with the two versions to test them: https://github.com/BBC-News/alephant-preview/commits/specify-renderer

In the Gemfile of the election-2016-renderer application, I added the github commit to the gem:
```
# gem "alephant-preview", :git => "https://github.com/BBC-News/alephant-preview.git", :ref => "f7528bca09eabd60a88941bc4f21d08b375c35dc"
gem "alephant-preview", :git => "https://github.com/BBC-News/alephant-preview.git", :ref => "46454921bd9382ae7636a49eaf34b0aabdab3b67"
```

The commit at the top is the 2.0.2 one, and the second one is the 2.0.1 one.
I ran:
```
bundle update
bundle exec alephant-preview
```

The update uses the renderer gem specified, and the latest of the other gems.

**Comment by samfrench** [comment]
The issue is around the double quotes and pattern matchers when using gsub. Fixed here: https://github.com/BBC-News/alephant-renderer/pull/19

**Pull request closed without merging**